src/grasp_generator.cpp

```diff
@@ -66,15 +66,16 @@ GraspGenerator::GraspGenerator(moveit_visual_tools::MoveItVisualToolsPtr visual_
   rosparam_shortcuts::get(parent_name, nh_, "translation_x_score_weight", translation_x_score_weight_);
   rosparam_shortcuts::get(parent_name, nh_, "translation_y_score_weight", translation_y_score_weight_);
   rosparam_shortcuts::get(parent_name, nh_, "translation_z_score_weight", translation_z_score_weight_);
+  rosparam_shortcuts::get(parent_name, nh_, "grasp_approach_vector_in_parent_frame",
+                          grasp_approach_vector_in_parent_frame_);
 
   // Set ideal grasp pose (currently only uses orientation of pose)
   ideal_grasp_pose_ = Eigen::Affine3d::Identity();
-  ideal_grasp_pose_ = ideal_grasp_pose_ * Eigen::AngleAxisd(M_PI / 2.0, Eigen::Vector3d::UnitZ()) *
-                      Eigen::AngleAxisd(-M_PI / 2.0, Eigen::Vector3d::UnitX());
+  ideal_grasp_pose_ = ideal_grasp_pose_ * Eigen::AngleAxisd(-M_PI, Eigen::Vector3d::UnitX());
```

Comment:
add lots of comments about how this was chosen and what its for

keep the old approach and make it a rosparam

something like...

int ideal_grasp_pose_type = 1 // Shelf picking
int ideal_grasp_pose_type = 2 // Bin picking